### ISSUE TITLE
Support GZipped files

### ISSUE THREAD

**Issue by odow** [odow/MathOptFormat.jl#39]
We probably want to support this across the board for all file formats.

**Comment by chriscoey** [comment]
ah yes, [CBLIB](http://cblib.zib.de/download/all/) (CBF library) has gzipped files.

**Comment by chriscoey** [comment]
let's brainstorm how to do this. 

for Hypatia benchmarks, I have need of a reader of `.cbf`, `.cbf.gz`, `.mof.json` and `.mof.json.gz` files. I want to have a folder with files of all these extensions jumbled up together, and i want to just read all of them one by one into MOI models. so I want to "dispatch" to the appropriate MOF or CBF read function based on the extension of the file string. similarly you could imagine wanting to dispatch to the correct write function based on the extension of the file string that you specify when writing. 

so, could we have read and write functions in `src/MathOptFormat.jl` that check the file extension of the string and then call gzip if necessary and call the appropriate format module read/write functions?

**Comment by chriscoey** [comment]
it would be convenient to have a function such as the following:
```
function read_into_model(filename::String)
    if endswith(filename, ".gz")
        io = GZip.open(filename, "r")
    else
        io = open(filename, "r")
    end
    if endswith(filename, ".cbf.gz") || endswith(filename, ".cbf")
        model = MathOptFormat.CBF.Model()
    elseif endswith(filename, ".mof.json.gz") || endswith(filename, ".mof.json")
        model = MathOptFormat.MOF.Model()
    else
        error("MathOptInterface.read_from_file is not implemented for this filetype: $filename")
    end
    MOI.read_from_file(model, io)
    return model
end
```
to handle gzipping and to create the right type of model (MOF, CBF etc).

**Comment by odow** [comment]
Happy to have that as a mathoptformat.jl function. Don't have to over load a moi function

**Comment by odow** [comment]
Closed by #40
